saw-core-what4/src/Verifier/SAW/Simulator/What4.hs

```diff
@@ -928,17 +948,23 @@ data TypedExpr sym where
   TypedExpr :: BaseTypeRepr ty -> SymExpr sym ty -> TypedExpr sym
 
 
+-- | Create a fresh constant with the given name and type.
+mkConstant ::
+  forall sym ty.
+  (IsSymExprBuilder sym) =>
+  sym -> String -> BaseTypeRepr ty -> IO (SymExpr sym ty)
+mkConstant sym name ty =
+  case W.userSymbol name of
```

Comment:
we should use `safeSymbol` here, to avoid the failure case

Reply:
47f626443ccd21e01557f3dc179d1c07d6adf33e